master/internal/db/postgres_experiments_intg_test.go

```diff
@@ -976,3 +976,123 @@ func validateExperimentMatch(t *testing.T, expected []*model.Experiment, actual
 	// map should be empty
 	require.Equal(t, len(m), 0)
 }
+
+func TestExperimentTotalStepTime(t *testing.T) {
+	ctx := context.Background()
+
+	require.NoError(t, etc.SetRootPath(RootFromDB))
+	db := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, db, MigrationsFromDB)
+
+	t.Run("invalid experiment, return 0.0, no error", func(t *testing.T) {
+
+		sec, err := db.ExperimentTotalStepTime(-1)
+		require.Equal(t, 0.0, sec)
+		require.NoError(t, err)
+	})
+
+	t.Run("experiment with single trial/task with null endtime", func(t *testing.T) {
+
+		user := RequireMockUser(t, db)
+		exp := RequireMockExperiment(t, db, user)
+		timeInSeconds, err := db.ExperimentTotalStepTime(exp.ID)
+		require.NoError(t, err)
+		require.Equal(t, 0.0, timeInSeconds)
+	})
+
+	t.Run("experiment with single trial/task with set endtime", func(t *testing.T) {
+
+		user := RequireMockUser(t, db)
+		exp := RequireMockExperiment(t, db, user)
+		_, task := RequireMockTrial(t, db, exp)
+		alloc := RequireMockAllocation(t, db, task.TaskID)
+		endTime := alloc.StartTime.Add(time.Hour)
+		alloc.EndTime = &endTime //its changing the mem not db
+		CompleteAllocation(ctx, alloc)
+		timeInSeconds, err := db.ExperimentTotalStepTime(exp.ID)
+		require.NoError(t, err)
+		require.Equal(t, 3600.0, timeInSeconds)
+	})
+
+	t.Run("experiment with multiple trials/tasks", func(t *testing.T) {
+
```

Comment:
```suggestion

```

this is a minor style comment but I probably wouldn't have added this new line after all the function definitions